packages/realm_dart/test/realm_value_test.dart

```diff
@@ -566,13 +568,186 @@ void main() {
         expect(obj.oneAny.type, RealmValueType.map);
         expectMatches(obj.oneAny, {'bool': true, 'double': 5.3});
 
+        writeIfNecessary(realm, () => obj.oneAny = RealmValue.from({'foo': 'bar'}));
+        expectMatches(obj.oneAny, {'foo': 'bar'});
+
         writeIfNecessary(realm, () => obj.oneAny = RealmValue.from(999));
         expectMatches(obj.oneAny, 999);
 
         writeIfNecessary(realm, () => obj.oneAny = RealmValue.from([1.23456789]));
         expectMatches(obj.oneAny, [1.23456789]);
       });
 
+      test('Map inside list when $managedString can be reassigned', () {
+        final realm = getMixedRealm();
+        final obj = AnythingGoes(
+            oneAny: RealmValue.from([
+          true,
+          {'foo': 'bar'},
+          5.3
+        ]));
+        if (isManaged) {
+          realm.write(() => realm.add(obj));
+        }
+
+        final list = obj.oneAny.asList();
+
+        writeIfNecessary(realm, () => list[1] = RealmValue.from({'new': 5}));
```

Comment:
What do you think about asserting/expecting that `list[1]` is a Realm map, right before reassigning to a new map. That would guard for potential manual changes to `oneAny` list at the top of the test and ensure that it's testing reassigning from one map to another.